router/payments.go

```diff
@@ -34,3 +37,44 @@ func (rt *Router) handlePaymentsBudget(w http.ResponseWriter, r *http.Request) {
 	}
 
 }
+
+func (rt *Router) handlePaymentSave(w http.ResponseWriter, r *http.Request) {
+	if r.Method != POST {
+		displayErrorPage(w, r, http.StatusMethodNotAllowed, "Method Not Allowed",
+			"The resource you requested does not support the method used.")
+	}
+
+	// Validate POST contents
+	bcID := strings.TrimSpace(r.PostFormValue("budget_category_id"))
+	date := strings.TrimSpace(r.PostFormValue("date"))
+	amount := strings.TrimSpace(r.PostFormValue("amount"))
+	note := strings.TrimSpace(r.PostFormValue("note"))
+
+	// Convert ID and amount
+	bcIDInt, err := strconv.Atoi(bcID)
+	if err != nil {
+		log.Fatalf("handlePaymentSave: %v\n", err)
+	}
+	amountFloat, err := strconv.ParseFloat(amount, 32)
+	if err != nil {
+		log.Fatalf("handlePaymentSave: %v\n", err)
+	}
+
+	data := struct {
+		Success      bool
+		ErrorMessage string
+	}{
+		Success: true,
+	}
+
+	_, err = rt.PaymentRepo.Create(date, float32(amountFloat), note, uint(bcIDInt))
```

Comment:
Is there a conv to `float32` directly? Also, AFAIK `float64` is the default float in Go.